tsfc/driver.py

```diff
@@ -196,12 +178,13 @@ def facetarea():
 
         # Check if the integral has a quad degree attached, otherwise use
         # the estimated polynomial degree attached by compute_form_data
-        quadrature_degree = params.get("quadrature_degree",
-                                       params["estimated_polynomial_degree"])
+        quad_degree = params.get("quadrature_degree")
+        if quad_degree in [None, "auto", "default", -1, "-1"]:
```

Comment:
That should be possible. I've done it only to avoid copying integral data and form data because I'm not sure I'm allowed to change their metadata.